plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -0,0 +1,581 @@
+  pretty VotingProcedure{..} =
+    vsep
+      [ "vpVote:" <+> pretty vpVote
+      , "vpAnchor:" <+> pretty vpAnchor
+      ]
+
+instance PlutusTx.Eq VotingProcedure where
+  {-# INLINEABLE (==) #-}
+  VotingProcedure a b == VotingProcedure a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data Committee = Committee
+  { committeeMembers :: Map ColdCommitteeCredential Haskell.Integer
+  -- ^ Committee members with epoch number when each of them expires
+  , committeeQuorum  :: PlutusTx.Rational
+  -- ^ Quorum of the committee that is necessary for a successful vote
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty Committee where
+  pretty Committee{..} =
+    vsep
+      [ "committeeMembers:" <+> pretty committeeMembers
+      , "committeeQuorum:" <+> pretty committeeQuorum
+      ]
+
+instance PlutusTx.Eq Committee where
+  {-# INLINEABLE (==) #-}
+  Committee a b == Committee a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+newtype ConstitutionHash = ConstitutionHash {getConstitutionHash :: PlutusTx.BuiltinByteString}
+  deriving stock (Generic)
+  deriving newtype
+    ( Haskell.Eq
+    , PlutusTx.Eq
+    , PlutusTx.Ord
+    , PlutusTx.Show
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+  deriving
+    ( -- | from hex encoding
+      IsString
+    , -- | using hex encoding
+      Haskell.Show
+    , -- | using hex encoding
+      Pretty
+    )
+    via V2.LedgerBytes
+
+data Constitution = Constitution
+  { constitutionHash   :: ConstitutionHash -- Balke2b_256 of some off-chain data
+  , constitutionScript :: Haskell.Maybe V2.ScriptHash
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty Constitution where
+  pretty Constitution{..} =
+    vsep
+      [ "constitutionHash:" <+> pretty constitutionHash
+      , "constitutionScript:" <+> pretty constitutionScript
+      ]
+
+instance PlutusTx.Eq Constitution where
+  {-# INLINEABLE (==) #-}
+  Constitution a b == Constitution a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data ProtocolVersion = ProtocolVersion
+  { pvMajor :: Haskell.Integer
+  , pvMinor :: Haskell.Integer
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty ProtocolVersion where
+  pretty ProtocolVersion{..} =
+    vsep
+      [ "pvMajor:" <+> pretty pvMajor
+      , "pvMinor:" <+> pretty pvMinor
+      ]
+
+instance PlutusTx.Eq ProtocolVersion where
+  {-# INLINEABLE (==) #-}
+  ProtocolVersion a b == ProtocolVersion a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data GovernanceAction
+  = -- TODO: this is currently empty.
+    ParameterChange
+  | -- | proposal to update protocol version
+    HardForkInitiation ProtocolVersion
+  | TreasuryWithdrawals (Map V2.Credential V2.Value)
+  | NoConfidence
+  | NewCommittee
+      [ColdCommitteeCredential] -- ^ Old committee
+      Committee -- ^ New Committee
+  | NewConstitution Constitution
```

Comment:
Sorry for the last minute changes, but we realized today that one piece is missing from here, namely the previous governance action id:

```suggestion
    ParameterChange GovernanceActionId
  | -- | proposal to update protocol version
    HardForkInitiation GovernanceActionId ProtocolVersion
  | TreasuryWithdrawals (Map V2.Credential V2.Value)
  | NoConfidence GovernanceActionId
  | NewCommittee
      GovernanceActionId
      [ColdCommitteeCredential] -- ^ Old committee
      Committee -- ^ New Committee
  | NewConstitution GovernanceActionId Constitution
```

`TreasuryWithdrawal` and `InfoAction` do not require one.